packages/cli/src/info/env_info.ts

```diff
@@ -0,0 +1,67 @@
+import * as os from 'node:os';
+import envinfo from 'envinfo';
+import { printer } from '../printer.js';
+import { EnvInfo } from './env_info_types.js';
+
+/**
+ * Provides environment information.
+ */
+export class EnvironmentInfoProvider {
+  /**
+   * Get environment information
+   */
+  async getEnvInfo(): Promise<EnvInfo> {
+    const info = await envinfo.run(
+      {
+        System: ['OS', 'CPU', 'Memory', 'Shell'],
+        Binaries: ['Node', 'Yarn', 'npm', 'pnpm'],
+        npmPackages: [
+          '@aws-amplify/backend',
+          '@aws-amplify/backend-cli',
+          'typescript',
+          'aws-cdk',
+          'aws-cdk-lib',
+          'aws-amplify',
+        ],
+        // npmGlobalPackages: ['@aws-amplify/cli'],
+      },
+      { json: true, showNotFound: true }
+    );
+    return JSON.parse(info);
+  }
+
+  /**
+   * Format environment information.
+   * @param info - The environment information.
+   * @returns The formatted environment information.
+   */
+  formatEnvInfo(info: EnvInfo): string {
```

Comment:
if this class is called `EnvironmentInfoProvider` then this method should be extracted out from here.
Can you add `class EnvironmentInfoFormatter` so we can have `environmentInforFormatter.format(envInvo)` ?

Reply:
Or similar to Kamil's comment above, if we don't have a need for unformatted output, just have `EnvironmentInfoProvider.getFormattedEnvInfo` or such.